### PR TITLE
AMQP should allow null AMQP app properties just like MQTT and HTTP. Properties with null value should not be dropped.

### PR DESCRIPTION
**Pull request by aklintsevich**
AMQP

### CODE REVIEW

**Comment by aklintsevich** [comment]
@vishal-gandhi @kapil-ms

**Comment by aklintsevich** [comment]
@davilu @timtay-microsoft @CIPop @abhipsaMisra @jenoola Can someone review this pull request?

File: iothub/device/src/MessageConverter.cs [hunk 1]
```diff
@@ -222,10 +222,6 @@ public static void UpdateAmqpMessageHeadersAndProperties(AmqpMessage amqpMessage
         public static bool TryGetAmqpObjectFromNetObject(object netObject, MappingType mappingType, out object amqpObject)
         {
             amqpObject = null;
-            if (netObject == null)
```
Comment (davilu) [inline iothub/device/src/MessageConverter.cs]: Shouldn't remove this check.

File: iothub/device/tests/MessageConverterTests.cs [hunk 2]
```diff
@@ -44,6 +44,7 @@ public void UpdateMessageHeaderAndPropertiesTest()
             
                 amqpMessage.ApplicationProperties.Map["Prop1"] = "Value1";
                 amqpMessage.ApplicationProperties.Map["Prop2"] = "Value2";
+                amqpMessage.ApplicationProperties.Map["Prop3"] = null;
```
Comment (davilu) [inline iothub/device/tests/MessageConverterTests.cs]: Shouldn't use `null` but `string.Empty`

Comment (aklintsevich) [reply iothub/device/tests/MessageConverterTests.cs]: This is testing that a null value does not get dropped.

**Comment by aklintsevich** [comment]
@davilu if MQTT and HTTP should be consistent with AMQP then this library should accept null values. We shouldn't be dropping app properties with null value. See image above.

File: iothub/device/tests/MessageConverterTests.cs (hunk 2 above)
Comment (davilu) [reply iothub/device/tests/MessageConverterTests.cs]: Shouldn't allow `null` value. Should use `string.Empty`.

**Comment by davilu** [comment]
`null` value has no standard serialization. Once this type of data received, it won't be able to tell it was set to`null` or empty string.

**Comment by aklintsevich** [comment]
If you do not want to allow null values, then you should not allow null values for MQTT and HTTP as well. If you would like to allow null values then you should allow them for all protocols rather than a subset.,

**Comment by aklintsevich** [comment]
MQTT and HTTP null values get serialized to empty string

**Comment by davilu** [comment]
Another question is it's not sure how `Amqp` library handle `null` value.

**Comment by aklintsevich** [comment]
I don't know how Amqp library will handle null values. I only tested the serialization of null values with empty string, which is consistent behavior with MQTT and HTTP protocols. When serializing with empty string there were no issues found when running IotHub BVTs.

**Comment by davilu** [comment]
We should try to fix HTTP/MQTT instead of AMQP. Pair with `null` value should be dropped. There are a lot of properties are ignore such as `Message.To`, `Message.ExpiryTimeUtc` etc. We shouldn't bring in new inconsistency.

**Comment by CIPop** [comment]
> I don't know how Amqp library will handle null values.
> When serializing with empty string there were no issues found when running IotHub BVTs.

@aklintsevich please create an issue describing the problem you are trying to solve or add an E2E test for us to understand the issue.

**Comment by aklintsevich** [comment]
https://github.com/Azure/azure-iot-sdk-csharp/issues/919

**Pull request closed without merging**